src/cluster/ArborX_DistributedDBSCAN.hpp

```diff
@@ -0,0 +1,133 @@
+/****************************************************************************
+ * Copyright (c) 2025, ArborX authors                                       *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DISTRIBUTED_DBSCAN_HPP
+#define ARBORX_DISTRIBUTED_DBSCAN_HPP
+
+#include <ArborX_DBSCAN.hpp>
+#include <detail/ArborX_AccessTraits.hpp>
+
+#include <Kokkos_Profiling_ScopedRegion.hpp>
+
+#include <mpi.h>
+
+namespace ArborX::Experimental
+{
+
+template <typename ExecutionSpace, typename Primitives, typename Coordinate,
+          typename Labels>
+void dbscan(MPI_Comm comm, ExecutionSpace const &exec_space,
+            Primitives const &primitives, Coordinate eps, int core_min_size,
+            Labels &labels,
```

Comment:
Changed the interface to take labels as opposed to returning as done with non-MPI version. Plan to change the non-MPI to also take them in.